_posts/2021-04-07-dynamic-import.md

```diff
@@ -0,0 +1,87 @@
+In this article we will delve into the dynamic import of JavaScript modules and the lazy loading of React components. We will examine through a real example how they work and how we can make our web application faster by reducing our initial bundle size. It is common to use TypeScript for its static type system. We often need types from our dependencies, but it can ruin our hardly achieved code splitting if we don't pay attention. I will show you a fairly new syntax to avoid it.
+
+## Dynamic import
+
+Dynamic import has reached the stage 4 of the TC39 process and is included in the ECMAScript 2020 language specification. Webpack, currently the most popular JavaScript module bundler, already support it since the v2 which was released in 2017. It makes it possible to load parts of your application at runtime. Maybe you use a heavy dependency only on specific cases or on a multi-language page you want to load only the desired localization files based on the user's preferences. This way you can make your site more performant and lightweight.
+
+The syntax of the dynamic import is quite simple, it just extend the the import keyword by making it possible to use it followed by parentheses with the path of your dependency in between.
+
+```javascript
+import('module/example').then(example => console.log(example.default)
+```
+
+> This sytax looks like a function call, but it is not. Import is not defined as a function, it is a specific operator.
+
+The code above loads the module at runtime and logs its default export to the console. This is just a basic example, you can use anything exported by the module in the callback function or load multiple modules at once with Promise.all.
+
+All popular modern bundlers support it and they automatically split dynamically imported modules to a separate bundle. All of the import statements of that module or dependency should be dynamic across your project to work as expected.
+
+## React.lazy
+
+It is also possible to import React components dynamically since React 16.6.  Reacz.lazy is a function which will handle your dynamic import and make a renderable React component from it. It has one parameter, which is a function returning the import:
```

Comment:
```suggestion
It is also possible to import React components dynamically since React 16.6.  React.lazy is a function which will handle your dynamic import and make a renderable React component from it. It has one parameter, which is a function returning the import:
```